tests/test_core/test_spectrum_group.py

```diff
@@ -169,68 +262,353 @@ def test_correct_radiation_pattern(self, spectrum_group_node):
         """ Return a df of calculated source info from the SpectrumGroup """
         # Apply pre-processing
-        out = spectrum_group_node.abs().ko_smooth()
+        out = spec_group_for_source_params.abs().ko_smooth()
+        # out = spec_group_for_source_params.abs().ko_smooth()
         out = out.correct_radiation_pattern()
         out = out.correct_spreading()
         out = out.correct_attenuation()
         out = out.correct_free_surface()
         # Return source df
         return out.calc_source_params()
 
+    @pytest.fixture(scope="function")
+    def source_params_preprocessing(self, spec_group_for_source_params) -> pd.DataFrame:
+        """
+        Return a df of calculated source info from the SpectrumGroup where
+        preprocessing occurred in the calculation call
+        """
+        return spec_group_for_source_params.calc_source_params(
+            apply_corrections=True, smoothing="ko_smooth"
+        )
+
+    @pytest.fixture(scope="function")
+    def source_params_no_preprocessing(
+        self, spec_group_for_source_params
+    ) -> pd.DataFrame:
+        """ Calculate source params without doing any kind of preprocessing """
+        with pytest.warns(UserWarning, match="has not been corrected"):
+            return spec_group_for_source_params.calc_source_params()
+
+    # Tests
+
     def test_basic(self, source_params):
         """ Ensure output is a nonempty DataFrame """
         assert isinstance(source_params, pd.DataFrame)
         assert not source_params.empty
 
+    def test_preprocessing(self, source_params, source_params_preprocessing):
+        """ Make sure that preprocessing was applied when desired """
+        np_assert(source_params_preprocessing, source_params, rtol=0.05)
+
+    def test_no_preprocessing(self, spectrum_group_node, source_params):
+        """ Make sure that preprocessing is not applied when not desired """
+        with pytest.warns(UserWarning, match="has not been corrected"):
+            df = spectrum_group_node.calc_source_params()
+        assert not np.allclose(df, source_params, rtol=0.05)
+
     def test_values(self, source_params):
         """ Ensure values are plausible """
         # This is intentionally somewhat brittle and intended to catch
         # subtle yet significant changes to the parameter calculations
         medians = source_params.median()
         checks = {
-            "fc": 5.7,
-            "omega0": 2.9e-3,
-            "velocity_squared_integral": 6.0e-3,
-            "moment": 5.8e12,
-            "potency": 1.4e2,
-            "energy": 2.4e6,
-            "mw": 2.5}
-        for key, val in checks.items():
-            np_assert(medians[key], val, rtol=0.02)
-
-
-class TestGroundMotionConversions:
+            "fc": (5.8, 0.02),  # (value, rtol)
+            "omega0": (6.35e-5, 0.12),
+            "moment": (1.3e11, 0.12),
+            "potency": (3.05, 0.12),
+            "energy": (1.17e7, 0.02),
+            "mw": (1.34, 0.03),
```

Comment:
So is this the same event? It dropped from 2.5 to 1.34 then?

Reply:
Yep. Pretty substantial change.